src/utilities/focus.ts

```diff
@@ -158,6 +159,61 @@ export function focusLastKeyboardFocusableNode(
   return false;
 }
 
+function getMenuFocusableDescendants(element: HTMLElement) {
+  return element.querySelectorAll(
+    MENUITEM_FOCUSABLE_SELECTORS,
+  ) as NodeListOf<HTMLElement>;
+}
+
+function getCurrentFocusedElementIndex(
+  allFocusableChildren: NodeListOf<HTMLElement>,
+  currentFocusedElement: HTMLElement,
+) {
+  let currentItemIdx = 0;
+
+  for (const focusableChild of allFocusableChildren) {
+    if (focusableChild === currentFocusedElement) {
+      break;
+    }
+    currentItemIdx++;
+  }
+  return currentItemIdx;
+}
+
+export function wrapFocusPreviousFocusableMenuItem(
+  parentElement: HTMLElement,
+  currentFocusedElement: HTMLElement,
+) {
+  const allFocusableChildren = getMenuFocusableDescendants(parentElement);
+  const currentItemIdx = getCurrentFocusedElementIndex(
+    allFocusableChildren,
+    currentFocusedElement,
+  );
+
+  if (currentItemIdx === 0) {
```

Comment:
Like the previous comment, the modulus version is `allFocusableChildren[
    (currentItemIdx - 1 + allFocusableChildren.length) %
      allFocusableChildren.length
  ].focus();`